MapboxNavigationTests/NavigationViewControllerTests.swift

```diff
@@ -0,0 +1,192 @@
+        
+        let delay = 10.0
+        let fireTime = DispatchTime.now() + delay
+        
+        DispatchQueue.main.asyncAfter(deadline: fireTime) {
+            
+            navigationViewController.mapView?.delegate?.mapView?(customMapView, didFinishLoading: customMapView.style!)
+            navigationViewController.mapView?.delegate?.mapViewDidFinishLoadingMap?(customMapView)
+            
+            routeController.locationManager(routeController.locationManager, didUpdateLocations: [turkStreetLocation])
+            
+            let wayNameView = (navigationViewController.mapViewController?.navigationView.wayNameView)!
+            let currentRoadName = wayNameView.text!
+            XCTAssertEqual(currentRoadName, roadName, "Expected: \(roadName); Actual: \(currentRoadName)")
+            XCTAssertTrue(wayNameView.isHidden, "WayNameView should be hidden.")
+            
+            mapviewExpectation.fulfill()
+        }
+        
+        wait(for: [mapviewExpectation], timeout: delay)
+    }
+    
+    // TODO: Most likely a UI-Test candidate.
+    func testNavigationViewControllerDelegateRoadNameAtLocationUmimplemented() {
+        
+        let dummyNavigationViewController = dependencies.dummyNavigationViewController
+        let routeController = dummyNavigationViewController.routeController!
+        
+        // Identify a location without a custom road name.
+        let fultonStreetLocation = dependencies.poi[2]
+        
+        let mapviewExpectation = expectation(description: "mapview_exp_default_roadname")
+        let customMapView = MGLMapView(frame: (dummyNavigationViewController.mapView?.bounds)!, styleURL: URL(string: "mapbox://styles/mapbox/streets-v9"))
+        
+        let delay = 1.0
+        let fireTime = DispatchTime.now() + delay
+        
+        DispatchQueue.main.asyncAfter(deadline: fireTime) {
+            
+            dummyNavigationViewController.mapView?.delegate?.mapView?(customMapView, didFinishLoading: customMapView.style!)
+            dummyNavigationViewController.mapView?.delegate?.mapViewDidFinishLoadingMap?(customMapView)
+            
+            routeController.locationManager(routeController.locationManager, didUpdateLocations: [fultonStreetLocation])
+            
+            // let roadName = "Fulton Street"
+            // let wayNameView = (dummyNavigationViewController.mapViewController?.navigationView.wayNameView)!
+            // let currentRoadName = wayNameView.text!
+            // XCTAssertEqual(currentRoadName, roadName, "Expected: \(roadName); Actual: \(currentRoadName)")
+            // XCTAssertFalse(wayNameView.isHidden, "WayNameView should be hidden.")
+            
+            mapviewExpectation.fulfill()
+        }
+        
+        wait(for: [mapviewExpectation], timeout: delay)
+    }
+}
+
+class NavigationViewControllerSpy: NavigationViewController { }
+class NavigationViewControllerFakeSpy: NavigationViewController { }
```

Comment:
Let's remove this, I don't think it's needed. also please see my comment below.

Reply:
Sure thing.